### PR TITLE
Changes sound synthesizers to be assemblies

### PR DESCRIPTION
**Pull request by unid15**
God help us all

Other changes:
- Rewrote sound synthesizer code to be better, fixing some minor bugs and annoyances
- Added sound synthesizers to all maps which previously didn't have it -- more have to be created using mechanics stuff
- Using a sound synthesizer on a mob now gives you a message (to hint at the effect)

### CODE REVIEW

File: code/game/objects/items/devices/megaphone.dm [hunk 1]
```diff
@@ -58,100 +58,8 @@
 	return
 
 
-
-
 /*
-	SOUND SYNTH BELOW THIS LINE
+	SOUND SYNTH NO LONGER LIVES HERE
 ----------------------------------------------
+	HE NOW LIVES IN code/modules/assembly/sound_synth.dm
```
Comment (PJB3005) [inline code/game/objects/items/devices/megaphone.dm]: Eh should probably just kill this comment whole.

megaphone.dm wasn't ever a good file for it anyways.

Not a place I'd look by far.

File: code/modules/assembly/sound_synth.dm [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+/obj/item/device/assembly/soundsynth/attack(mob/living/M as mob, mob/living/user as mob, def_zone)
+	if(M == user) //If you target yourself, cycle sound forward
+		return src.CycleForward()
+	else
+		if(spam_flag + 20 < world.timeofday)
+			if(sounds[sound_flag+1])
+				var/sound_to_play = sounds[sound_flag+1]
+
+				spam_flag = world.timeofday
+				M << sound(sounds[sound_to_play])
```
Comment (PJB3005) [inline code/modules/assembly/sound_synth.dm]: I don't see any message to `M` here saying he's being messed with, adding one might be a good idea.

File: code/game/objects/items/devices/megaphone.dm (hunk 1 above)
Comment (ComicIronic) [inline code/game/objects/items/devices/megaphone.dm]: Yeah, no need for this.

**Comment by FudgePucker** [comment]
This is missing the 'HONK' tag.
Also merge when?

File: code/modules/assembly/sound_synth.dm [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+*/
+/obj/item/device/assembly/soundsynth/verb/CycleBackward()
+	set category = "Object"
+	set name = "Cycle Sound Backward"
+
+	sound_flag--
+	if(sound_flag < 0)
+		sound_flag = sounds.len - 1
+
+	if(sounds[sound_flag+1])
```
Comment (clusterfack) [inline code/modules/assembly/sound_synth.dm]: Is this right? It seems like it shouldn't be the same for both cycle forward and backward

**Comment by ihadtoregisterforthis** [comment]
But muh mime gimmick item!
With this every MoMMI and cargonian will spam explosion sounds.
At least make the bike horn made from the protolathe.

**Comment by PJB3005** [comment]
Eh if it's gonna be spawned maybe make it buildable with the reverse engine and the mechanics fabricator.

File: code/modules/assembly/sound_synth.dm [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+/*
+This is to cycle sounds forward
+*/
+/obj/item/device/assembly/soundsynth/verb/CycleForward()
+	set category = "Object"
+	set name = "Cycle Sound Forward"
+	//writepanic("[__FILE__].[__LINE__] ([src.type])([usr ? usr.ckey : ""]) \\/obj/item/device/soundsynth/verb/CycleForward()  called tick#: [world.time]")
+	sound_flag++
+	if(sound_flag >= sounds.len)
+		sound_flag = 0
```
Comment (ComicIronic) [inline code/modules/assembly/sound_synth.dm]: Wait, did you just make this 0 and then add 1 everywhere it's used?
WHY?

**Pull request closed without merging**